pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,421 @@
+	ret := []jobrunaggregatorapi.JobRow{}
+	for i := range allJobs {
+		curr := allJobs[i]
+		if jobNames.Has(curr.JobName) {
+			ret = append(ret, curr)
+		}
+	}
+	return ret
+}
+
+// TestCaseChecker checks if a test passes certain criteria across all job runs
+type TestCaseChecker interface {
+	// CheckTestCase returns a test case based on whether a test has passed certain criteria across job runs
+	CheckTestCase(ctx context.Context, jobRuns []jobrunaggregatorapi.JobRunInfo) *junit.TestCase
+}
+
+type minimumRequiredPassesTestCaseChecker struct {
+	id                     testIdentifier
+	testNameSuffix         string
+	requiredNumberOfPasses int
+}
+
+type testStatus int
+
+const (
+	testSkipped testStatus = iota
+	testPassed
+	testFailed
+)
+
+func getTestStatus(id testIdentifier, testSuite *junit.TestSuite) testStatus {
+	if len(id.testSuites) == 0 || id.testSuites[0] != testSuite.Name {
+		return testSkipped
+	}
+	// We have a top level suite match, search for test case
+	if len(id.testSuites) == 1 {
+		for _, testCase := range testSuite.TestCases {
+			// Exact match will result in either pass or fail
+			if testCase.Name == id.testName {
+				if testCase.FailureOutput == nil {
+					return testPassed
+				} else {
+					return testFailed
+				}
+			}
+		}
+		return testSkipped
+	}
+	// Search next level
+	next := id
+	next.testSuites = id.testSuites[1:]
+	for _, childSuite := range testSuite.Children {
+		status := getTestStatus(next, childSuite)
+		if status == testPassed || status == testFailed {
+			return status
+		}
+	}
+	return testSkipped
+}
+
+func (r minimumRequiredPassesTestCaseChecker) addTestResultToDetails(currDetails *jobrunaggregatorlib.TestCaseDetails,
+	jobRun jobrunaggregatorapi.JobRunInfo, status testStatus) {
+	switch status {
+	case testPassed:
+		currDetails.Passes = append(
+			currDetails.Passes,
+			jobrunaggregatorlib.TestCasePass{
+				JobRunID:       jobRun.GetJobRunID(),
+				HumanURL:       jobRun.GetHumanURL(),
+				GCSArtifactURL: jobRun.GetGCSArtifactURL(),
+			})
+	case testFailed:
+		currDetails.Failures = append(
+			currDetails.Failures,
+			jobrunaggregatorlib.TestCaseFailure{
+				JobRunID:       jobRun.GetJobRunID(),
+				HumanURL:       jobRun.GetHumanURL(),
+				GCSArtifactURL: jobRun.GetGCSArtifactURL(),
+			})
+	default:
+		currDetails.Skips = append(
+			currDetails.Skips,
+			jobrunaggregatorlib.TestCaseSkip{
+				JobRunID:       jobRun.GetJobRunID(),
+				HumanURL:       jobRun.GetHumanURL(),
+				GCSArtifactURL: jobRun.GetGCSArtifactURL(),
+			})
+	}
+}
+
+// CheckTestCase returns a test case based on whether a test has passed certain criteria across job runs
+func (r minimumRequiredPassesTestCaseChecker) CheckTestCase(ctx context.Context, jobRuns []jobrunaggregatorapi.JobRunInfo) *junit.TestCase {
+	testName := fmt.Sprintf("test '%s.%s' has required number of successful passes across payload jobs", strings.Join(r.id.testSuites, jobrunaggregatorlib.TestSuitesSeparator), r.id.testName)
```

Comment:
I'm a little surprised that we don't keep the suite matching.  Doesn't aggregation normally keep the suites matching?